ingest/sqs/pools/transformer/pool_transformer.go

```diff
@@ -230,51 +209,138 @@ func (pi *poolTransformer) convertPool(
 	pool = pool.AsSerializablePool()
 
 	// filtered balances consisting only of the pool denom balances.
-	filteredBalances := sdk.NewCoins()
+	filteredBalances := filterBalances(balances, poolDenomsMap)
```

Comment:
Note: the following are the changes done below
- Extract separate functions for (The logic is kept the same, simply new functions for readability and testability)
  * creating pool denoms map
  * creating filtered balances
  * compute pool liquidity capitalization in uosmo

Added logic:
- Convert uosmo liquidity capitalization to USDC and descale using the precision scaling factor